scripts/watchBackedUpNode.js

```diff
@@ -0,0 +1,132 @@
+// Validator node failover script
+//
+// The script should be run on the secondary (failover) node that backs up a
+// primary node. The secondary node has signing disabled by default. It starts
+// signing blocks when the primary node is a validator but has not produced a
+// block when it was its turn and this script cannot connect to it by HTTP. The
+// secondary node stops sigining blocks as soon as the HTTP connection to the
+// primary node re-establishes.
+
+var assert = require("assert");
+var fs = require("fs");
+var { promisify } = require("util");
+var readFile = promisify(fs.readFile);
+var ethers = require("ethers");
+
+const PORT1 = "8543";
+const PORT2 = "8546";
+const RETRY_TIMEOUT_SECONDS = 2;
+const SCAN_INTERVAL_SECONDS = 5;
+const ARTIFACTS_PATH = "../posdao-contracts/build/contracts/";
+const PASSWORD_PATH = "/../config/password"
+const SIGNER_ADDRESS1 = "0x522df396ae70a058bd69778408630fdb023389b2";
+const SIGNER_ADDRESS2 = "0x522df396ae70a058bd69778408630fdb023389b2";
+const DUMMY_SIGNER_ADDRESS = "0x69103d3decf4a462e7f95b36684b04f86a950028";
+
+var Web3 = require("web3");
+var web3_1 = new Web3(new Web3.providers.HttpProvider(`http://localhost:${PORT1}`));
+var web3_2 = new Web3(new Web3.providers.HttpProvider(`http://localhost:${PORT2}`));
+var provider = new ethers.providers.JsonRpcProvider(`http://localhost:${PORT2}`);
+// `true` if the primary is required to sign and `false` if the secondary does.
+var primaryHasToSign = true;
+var validatorSetContract = new web3_2.eth.Contract(
+  require(`${ARTIFACTS_PATH}ValidatorSetAuRa.json`).abi,
+  '0x1000000000000000000000000000000000000001'
+);
+
+async function scanBlocks(depth) {
+    assert(typeof depth === "number");
+    var lastBlock = await web3_2.eth.getBlock("latest");
+    var lastBlockNum = lastBlock.number;
+    assert(typeof lastBlockNum === "number");
+    if (lastBlockNum < depth) {
+        return true;
+    }
+    var startBlockNum = lastBlockNum - depth;
+
+    console.log(`Scanning blocks from ${startBlockNum} to ${lastBlockNum}`);
+
+    for (var i = startBlockNum;  i <= lastBlockNum; i++) {
+        let block = await web3_2.eth.getBlock(i);
+        if (block.author === SIGNER_ADDRESS) {
```

Comment:
Should be `SIGNER_ADDRESS1` probably

Reply:
I decided to remove `SIGNER_ADDRESS2` instead because both signer addresses were the same.